api/tests/identity-access-management/integration/domain/usecases/register-lti-platform.test.js

```diff
@@ -0,0 +1,263 @@
+
+    const expectedPixConfig = {
+      response_types: ['id_token'],
+      jwks_uri: 'https://api.test.pix.fr/api/lti/keys',
+      initiate_login_uri: 'https://api.test.pix.fr/api/lti/init',
+      grant_types: ['client_credentials', 'implicit'],
+      redirect_uris: ['https://api.test.pix.fr/api/lti/launch'],
+      application_type: 'web',
+      token_endpoint_auth_method: 'private_key_jwt',
+      client_name: 'Pix',
+      logo_uri: 'https://app.pix.fr/images/pix-logo.svg',
+      scope: [
+        'https://purl.imsglobal.org/spec/lti-ags/scope/score',
+        'https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly',
+        'https://purl.imsglobal.org/spec/lti-ags/scope/lineitem',
+        'https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly',
+      ].join(' '),
+      'https://purl.imsglobal.org/spec/lti-tool-configuration': {
+        target_link_uri: 'https://api.test.pix.fr/api/lti',
+        domain: 'api.test.pix.fr',
+        description: 'Intégration avec la plateforme Pix',
+        messages: [
+          {
+            type: 'LtiDeepLinkingRequest',
+            target_link_uri: 'https://api.test.pix.fr/api/lti/content-selection',
+          },
+        ],
+        claims: ['sub', 'iss', 'name', 'family_name', 'given_name', 'email'],
+      },
+    };
+
+    const publicKey = { kty: 'test' };
+    const privateKey = Symbol('privateKey');
+    const encryptedPrivateKey = 'encryptedPrivateKey';
+    const expectedLtiPlatformRegistration = {
+      clientId: toolConfig.client_id,
+      platformOrigin,
+      toolConfig,
+      platformOpenIdConfigUrl,
+      publicKey,
+      encryptedPrivateKey,
+    };
+
+    const httpAgent = {
+      get: sinon.stub().resolves({ code: 200, isSuccessful: true, data: platformOpenIdConfig }),
+      post: sinon.stub().resolves({ code: 201, isSuccessful: true, data: toolConfig }),
+    };
+
+    const cryptoService = {
+      generateJSONWebKeyPair: sinon.stub().resolves({ privateKey, publicKey }),
+      encrypt: sinon.stub().resolves(encryptedPrivateKey),
+    };
+
+    const dependencies = { httpAgent, ltiPlatformRegistrationRepository, cryptoService };
+
+    // when
+    await usecases.registerLtiPlatform({
+      platformConfigurationUrl: platformOpenIdConfigUrl,
+      registrationToken: 'token1234',
+      ...dependencies,
+    });
+
+    // then
+    expect(httpAgent.get).to.have.been.calledWith({
+      url: platformOpenIdConfigUrl,
+      headers: { Accept: 'application/json' },
+    });
+    expect(httpAgent.post).to.have.been.calledWith({
+      url: platformOpenIdConfig.registration_endpoint,
+      headers: {
+        Authorization: 'Bearer token1234',
+        'Content-type': 'application/json',
+      },
+      payload: expectedPixConfig,
+    });
+
+    expect(cryptoService.generateJSONWebKeyPair).to.have.been.calledOnce;
+    expect(cryptoService.encrypt).to.have.been.calledWith(JSON.stringify(privateKey));
```

Comment:
Normalement dans un test d'intégration rien n'est stubbé. Mais là comme il y a des appels réseau ça se comprend. Mais on n'a pas à faire ce type d'assertion dans un test d'intégration :

```suggestion
    expect(cryptoService.generateJSONWebKeyPair).to.have.been.calledOnce;
    expect(cryptoService.encrypt).to.have.been.calledWith(JSON.stringify(privateKey));
```

Reply:
C'est la seule chose qui n'est pas testée dans le test d'acceptance. En effet, il faudrait récupérer les 2 clés (l'une chiffrée et l'autre en clair), et vérifier à coup de déchiffrement, signature, et vérif de signature que ce qui est présent dans les champs de BDD est ok. Ça nous semblait très lourd...
On pourrait aussi faire un test unitaire juste pour ce point, mais je ne suis pas convaincue que ça en vaille la peine.

Reply:
Bien reçu. D'accord. Discussion résolue.